misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm

```diff
@@ -23,126 +24,145 @@ use strict;
+  my $species_id = $ref_arg->{species_id};
+  my $files      = $ref_arg->{files};
+  my $verbose    = $ref_arg->{verbose} // 0;
+  my $dbi        = $ref_arg->{dbi} // $self->dbi;
+
+  if ( ( !defined $source_id )
+    or ( !defined $species_id )
+    or ( !defined $files ) )
+  {
     croak "Need to pass source_id, species_id and file as pairs";
   }
-  $verbose |=0;
 
-  my $file = @{$files}[0];
+  my $file = shift @{$files};
 
   my %label;
   my %version;
   my %description;
   my %accession;
 
-  my $sql = 'select source_id, priority_description from source where name like "MGI"';
+  my $sql = 'select source_id from source where name like "MGI"';
   my $sth = $dbi->prepare($sql);
 
   $sth->execute();
-  my ($mgi_source_id, $desc);
-  $sth->bind_columns(\$mgi_source_id, \$desc);
+  my ($mgi_source_id);
+  $sth->bind_columns( \$mgi_source_id );
   my @arr;
-  while($sth->fetch()){
+  while ( $sth->fetch() ) {
     push @arr, $mgi_source_id;
   }
   $sth->finish;
 
-  $sql = "select accession, label, version,  description from xref where source_id in (".join(", ",@arr).")";
+  $sql =
+"select accession, label, version,  description from xref where source_id in ("
+    . join( ", ", @arr ) . ")";
 
   $sth = $dbi->prepare($sql);
   $sth->execute();
-  my ($acc, $lab, $ver);
-  $sth->bind_columns(\$acc, \$lab, \$ver, \$desc);
-  while (my @row = $sth->fetchrow_array()) {
-    if(defined($desc)){
-      $accession{$lab} = $acc;
-      $label{$acc} = $lab;
-      $version{$acc} = $ver;
+  my ( $acc, $lab, $ver, $desc );
+  $sth->bind_columns( \$acc, \$lab, \$ver, \$desc );
+  while ( my @row = $sth->fetchrow_array() ) {
+    if ( defined($desc) ) {
+      $accession{$lab}   = $acc;
+      $label{$acc}       = $lab;
+      $version{$acc}     = $ver;
       $description{$acc} = $desc;
     }
   }
   $sth->finish;
 
-
-
   #
   # Get master xref ids via the ccds label.
   #
 
-  $sql = 'select x.label, x.xref_id from xref x, source s where x.source_id = s.source_id and s.name ="CCDS"';
-  
+  $sql =
+'select x.label, x.xref_id from xref x, source s where x.source_id = s.source_id and s.name ="CCDS"';
+
   my %ccds_label_to_xref_id;
   $sth = $dbi->prepare($sql);
   $sth->execute();
   my ($xref_id);
-  $sth->bind_columns(\$lab, \$xref_id);
-  while (my @row = $sth->fetchrow_array()) {
-    $ccds_label_to_xref_id{$row[0]} = $row[1];
+  $sth->bind_columns( \$lab, \$xref_id );
+
+  while ( my @row = $sth->fetchrow_array() ) {
+    $ccds_label_to_xref_id{ $row[0] } = $row[1];
   }
   $sth->finish;
 
-
-
-  my $ua = LWP::UserAgent->new();
-  $ua->timeout(10);
-  $ua->env_proxy();
-  
-
-  my $count = 0;
-  my $ccds_missing = 0;
+  my $count              = 0;
+  my $ccds_missing       = 0;
   my $entrezgene_missing = 0;
 
   my $mgi_io = $self->get_filehandle($file);
   if ( !defined $mgi_io ) {
-    print STDERR "ERROR: Could not open $file\n";
-    return 1;    # 1 is an error
+    croak "ERROR: Could not open $file\n";
```

Comment:
As discussed in other pull requests, if we're generating an error, we don't need ERROR in the message.